packages/backend/src/apps/m365-excel/common/get-transfer-details.ts

```diff
@@ -0,0 +1,63 @@
+import { IGlobalVariable, ITransferDetails } from '@plumber/types'
+
+import { getM365TenantInfo } from '@/config/app-env-vars/m365'
+import { getEmptyConnectionDetails } from '@/helpers/get-basic-transfer-details'
+
+import { AuthData } from './auth-data'
+
+async function getTransferDetails(
+  $: IGlobalVariable,
+): Promise<ITransferDetails> {
+  const authData = $.auth?.data as AuthData
+  if (!authData) {
+    return getEmptyConnectionDetails($.step.position, $.app.name)
+  }
+
+  if (!authData.tenantKey) {
+    return {
+      position: $.step.position,
+      appName: $.app.name,
+      connectionName: 'Invalid connection: Missing tenant',
+    }
+  }
+
+  if (!authData.folderId) {
+    return {
+      position: $.step.position,
+      appName: $.app.name,
+      connectionName: 'Invalid connection: Missing folder',
+    }
+  }
+
+  // TODO (mal/WL): check if there is a better approach and if the table needs to be shown too
+  const fileId = $.step.parameters.fileId as string
+  if (!fileId) {
+    return {
+      position: $.step.position,
+      appName: $.app.name,
+      connectionName: 'No M365 file is selected',
+    }
+  }
+
+  // query takes around 0.5s, is this worth?
```

Comment:
yes, definitely. Let's see if we can build some sort of cache later. This will bite us when we scale up.

Reply:
let me put a todo here then